src/SourceLocator/SourceStubber/ReflectionSourceStubber.php

```diff
@@ -120,6 +121,13 @@ public function generateFunctionStub(string $functionName): ?StubData
         $this->addDocComment($functionNode, $functionReflection);
         $this->addParameters($functionNode, $functionReflection);
 
+        $returnType = $functionReflection->getReturnType();
+        assert($returnType instanceof CoreReflectionNamedType || $returnType instanceof CoreReflectionUnionType || $returnType === null);
```

Comment:
This assertion goes away if we make the type stricter

Reply:
`$functionReflection` is PHP `ReflectionFunction` and it returns `?ReturnType` (https://www.php.net/manual/en/reflectionfunctionabstract.getreturntype.php) so the assert is needed to make PHPStan and Psalm happy.